### PR TITLE
feat(click tracking): make click tracking work better by going to the…

### PR DESCRIPTION
**Pull request by rileylnapier**
… new stack

we were piggybacking off the old click redirect tracking and that is slow, this shold make things faster

## Description

- support and subscribe to new action format that has trackingId in `data` vs using redirect urls for both inbox and toast
- deprecate rehydrate messages action in favor of using "init"
- don't refetch message count within an hour to help cache

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> `Fix [#1]()`

### CODE REVIEW

File: packages/react-hooks/src/inbox/__tests__/reducer.spec.ts [hunk 1]
```diff
@@ -118,26 +114,6 @@ describe("inbox reducer", () => {
     });
   });
 
-  describe(`action ${INBOX_REHYDRATE_MESSAGES}`, () => {
```
Comment (rileylnapier) [inline packages/react-hooks/src/inbox/__tests__/reducer.spec.ts:121]: we just rely on the "init" action to rehydrate

File: packages/react-hooks/src/inbox/middleware.ts [hunk 2]
```diff
@@ -10,7 +10,7 @@ export default (api) => (store) => (next) => async (action) => {
   const state = store.getState();
 
   switch (action.type) {
-    case "root/PAGE_VISIBLE":
+    //case "root/PAGE_VISIBLE":
```
Comment (rileylnapier) [inline packages/react-hooks/src/inbox/middleware.ts:13]: i think the page visible, which was leaving a tab and coming back may have caused more issues that it solved

File: packages/react-hooks/src/inbox/use-inbox-actions.ts [hunk 3]
```diff
@@ -102,9 +99,19 @@ const useInboxActions = (): IInboxActions => {
...
+    if (!fromWS) {
+      await inboxClient.markRead(messageId);
+    }
+  };
+
   const handleInit: IInboxActions["init"] = async (payload) => {
     dispatch(initInbox(payload));
-    handleGetUnreadMessageCount();
+    const dateDiff = getDateDiff(payload.lastMessagesFetched);
+    if (!dateDiff || dateDiff > 3600000) {
```
Comment (rileylnapier) [inline packages/react-hooks/src/inbox/use-inbox-actions.ts:112]: only refetch unread count on reload if its been an hour

File: packages/react-inbox/src/components/Inbox/index.tsx [hunk 4]
```diff
@@ -132,28 +132,28 @@ const Inbox: React.FunctionComponent<InboxProps> = (props) => {
...
   };
 
+  const localStorageState = useLocalStorageMessages(
+    courierContext.clientKey,
+    courierContext.userId
+  );
+
   useEffect(() => {
     init({
+      ...localStorageState,
```
Comment (rileylnapier) [inline packages/react-inbox/src/components/Inbox/index.tsx:142]: this replaced "rehydrate state"

File: packages/react-hooks/src/inbox/use-inbox-actions.ts (hunk 3 above)
Comment (drew-y) [reply packages/react-hooks/src/inbox/use-inbox-actions.ts:105]: Could `dateDiff` be zero? If so, `!dateDiff` would eval to `true`. Which, based on the `dateDiff > 3600000` might be undesirable.

Might be safer to do:
```suggestion
    if (dateDiff === undefined || dateDiff > 3600000) {
```

File: packages/react-inbox/src/components/Inbox/index.tsx [hunk 5]
```diff
@@ -132,28 +132,28 @@ const Inbox: React.FunctionComponent<InboxProps> = (props) => {
...
     if (!isOpen) {
-      const now = new Date().getTime();
-      const dateDiff = lastMessagesFetched
-        ? now - lastMessagesFetched
-        : undefined;
-
       setView("messages");
 
+      const dateDiff = getDateDiff(lastMessagesFetched);
       if (!dateDiff || dateDiff > 3600000) {
```
Comment (drew-y) [inline packages/react-inbox/src/components/Inbox/index.tsx:157]: Here too.

File: packages/react-inbox/src/components/Messages2.0/Message.tsx [hunk 6]
```diff
@@ -270,20 +260,43 @@ const MessageWrapper: React.FunctionComponent<
...
-        };
+        window.open(clickActionDetails?.href, "_blank");
+      } else {
+        window.location.href = clickActionDetails?.href;
       }
-    }
+    };
 
-    containerProps.onMouseDown = handleClickMessage;
+    //containerProps.onMouseDown = handleClickMessage;
```
Comment (drew-y) [inline packages/react-inbox/src/components/Messages2.0/Message.tsx:299]: Should this still be commented out?

**Review by drew-y — APPROVED** [review]
A few non-blocking questions / comments. Looks good!

**Review by drew-y — DISMISSED** [review]

**Review by drew-y — APPROVED** [review]

**Pull request merged**